File: 2020/python/day1/main.py
from aoc_utils import run_with_timer

data = [int(x.strip()) for x in open("input.txt").readlines()]


def part_one():
	return next(first*second for i, first in enumerate(data) for second in data[i+1:] if first + second == 2020)


def part_two():
	return next(first*second*third for i, first in enumerate(data) for j, second in enumerate(data[i+1:]) for third in data[j+1:] if first + second + third == 2020)


if __name__ == '__main__':
	run_with_timer(part_one)  # 197451 -- took 0 ms
	run_with_timer(part_two)  # 138233720 -- took 94 ms

File: 2020/golang/src/day1/main.go
package main

import (
	"fmt"
	"strconv"
	"utils"
)

func getData(lines []string) []int {
	var data []int
	for _, s := range lines {
		i, _ := strconv.Atoi(s)
		data = append(data, i)
	}
	return data
}

func partOne(data []int) int {
	for i := 0; i < len(data)-1; i++ {
		for j := i + 1; j < len(data); j++ {
			if data[i]+data[j] == 2020 {
				return data[i] * data[j]
			}
		}
	}
	return 0
}

func partTwo(data []int) int {
	for i := 0; i < len(data)-2; i++ {
		for j := i + 1; j < len(data)-1; j++ {
			for k := j + 1; k < len(data); k++ {
				if data[i]+data[j]+data[k] == 2020 {
					return data[i] * data[j] * data[k]
				}
			}
		}
	}
	return 0
}

func main() {
	var data = getData(utils.GetLines("input.txt"))
	fmt.Println(partOne(data))
	fmt.Println(partTwo(data))
}
